### DISCUSSION TITLE
Search AttachAction records on relationship?

### DISCUSSION THREAD

**Discussion by ProgrammerZ** [filamentphp/filament#14531] [Help]
### Package

Panel builder

### Package Version

v3.2.115

### How can we help you?

Is there a way to search `AttachAction` record on relationship data?

```
AttachAction::make('contact')
    ->form(fn (AttachAction $action): array => [
        $action->getRecordSelect()
    ])
->recordSelectSearchColumns(['phone.number'])`
```
Any workaround or I need to create custom Action?

**Comment by intrepidws** [comment]
Did you ever find an answer here?

**Reply by ProgrammerZ** [reply]
@intrepidws unfortunately not. Overall I found hard to customize relationship implementations in Filament, so now I almost always choose to go with custom Actions.

**Reply by intrepidws** [reply]
It feels like we need some way to customize the base query itself, so a `->with('phone')` could be added. Pretty sure with that, your implementation above would work.

**Reply by ProgrammerZ** [reply]
@intrepidws I believe I tried that one with `recordSelectOptionsQuery()` with no luck. If you find a solution, please post. 
I had to abandoned `AttachAction`, due to the fact that they don't support search query customization as with regular Form input and `searchable()`.

**Reply by intrepidws** [reply]
I tried the same thing and that didn't work.

I think the only answer currently is to use a [virtual column](https://filamentphp.com/docs/3.x/forms/fields/select#customizing-the-relationship-option-labels).
